pkg/commands/workflows/plan_status_comments.go

```diff
@@ -0,0 +1,225 @@
+		if c.GitHubFlags.FlagGitHubOwner == "" {
+			merr = errors.Join(merr, fmt.Errorf("missing flag: github-owner is required"))
+		}
+
+		if c.GitHubFlags.FlagGitHubRepo == "" {
+			merr = errors.Join(merr, fmt.Errorf("missing flag: github-repo is required"))
+		}
+
+		if c.flagPullRequestNumber <= 0 {
+			merr = errors.Join(merr, fmt.Errorf("missing flag: pull-request-number is required"))
+		}
+
+		if c.flagInitResult == "" {
+			merr = errors.Join(merr, fmt.Errorf("missing flag: init-result is required"))
+		}
+
+		if c.flagPlanResult == "" {
+			merr = errors.Join(merr, fmt.Errorf("missing flag: plan-result is required"))
+		}
+
+		return merr
+	})
+
+	return set
+}
+
+func (c *PlanStatusCommentsCommand) Run(ctx context.Context, args []string) error {
+	logger := logging.FromContext(ctx)
+
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	parsedArgs := f.Args()
+	if len(parsedArgs) > 0 {
+		return flag.ErrHelp
+	}
+
+	c.actions = githubactions.New(githubactions.WithWriter(c.Stdout()))
+	actionsCtx, err := c.actions.Context()
+	if err != nil {
+		return fmt.Errorf("failed to load github context: %w", err)
+	}
+
+	c.cfg = &PlanStatusCommentsConfig{}
+	if err := c.cfg.MapGitHubContext(actionsCtx); err != nil {
+		return fmt.Errorf("failed to load github context: %w", err)
+	}
+	logger.DebugContext(ctx, "loaded configuration", "plan_status_comments_config", c.cfg)
+
+	c.gitHubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithRetryInitialDelay(c.RetryFlags.FlagRetryInitialDelay),
+		github.WithRetryMaxAttempts(c.RetryFlags.FlagRetryMaxAttempts),
+		github.WithRetryMaxDelay(c.RetryFlags.FlagRetryMaxDelay),
+	)
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for the Guardian remove plan comments process.
+func (c *PlanStatusCommentsCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx).
+		With("github_owner", c.GitHubFlags.FlagGitHubOwner).
+		With("github_repo", c.GitHubFlags.FlagGitHubOwner).
+		With("init_result", c.flagInitResult).
+		With("plan_result", c.flagPlanResult)
+
+	logger.DebugContext(ctx, "determining plan status...")
+
+	c.gitHubLogURL = fmt.Sprintf("[[logs](%s/%s/%s/actions/runs/%d/attempts/%d)]", c.cfg.ServerURL, c.GitHubFlags.FlagGitHubOwner, c.GitHubFlags.FlagGitHubRepo, c.cfg.RunID, c.cfg.RunAttempt)
+	logger.DebugContext(ctx, "computed github log url", "github_log_url", c.gitHubLogURL)
+
+	// this case improves user experience, when the planning job does not have a plan diff
+	// there are no comments, this informs the user that the job ran successfully
+	if c.flagInitResult == "success" && c.flagPlanResult == "success" {
+		if _, err := c.gitHubClient.CreateIssueComment(
+			ctx,
+			c.GitHubFlags.FlagGitHubOwner,
+			c.GitHubFlags.FlagGitHubRepo,
+			c.flagPullRequestNumber,
+			fmt.Sprintf("%s 🟩 Plan completed successfully. %s", plan.CommentPrefix, c.gitHubLogURL),
```

Comment:
I thought we print out the plan diff?

Reply:
We do that in the plan step, this is a follow up job. See the PR description.